controller/aks-cluster-config-handler.go

```diff
@@ -554,10 +562,18 @@ func (h *Handler) GetClusterKubeConfig(ctx context.Context, spec *aksv1.AKSClust
 
 // buildUpstreamClusterState creates an AKSClusterConfigSpec (spec for the AKS cluster state) from the existing
 // cluster configuration.
-func (h *Handler) BuildUpstreamClusterState(ctx context.Context, spec *aksv1.AKSClusterConfigSpec) (*aksv1.AKSClusterConfigSpec, error) {
+func BuildUpstreamClusterState(ctx context.Context, secretsCache wranglerv1.SecretCache, secretClient wranglerv1.SecretClient, spec *aksv1.AKSClusterConfigSpec) (*aksv1.AKSClusterConfigSpec, error) {
```

Comment:
can we pass cluster client instead of creating it on every reconcile?

Reply:
yes, we can pass cluster client but it will break once again Rancher integration https://github.com/rancher/rancher/blob/release/v2.7/pkg/controllers/management/clusterupstreamrefresher/aks_upstream_spec.go#L14

I would keep it as it is for now, we want to unblock fixes for customer issue first and then we can continue with refactoring

Reply:
We can create the client somewhere in core rancher and pass it to this function. Let's merge the PR now and refactor it when we release the fix.